Config namespace fix and changelog update (#477)
canonical/craft-application: docs/reference/changelog.rst

Changelog
*********

4.2.3 (2024-Sep-18)
-------------------

Application
===========

> - get_arg_or_config now correctly checks the config service if the passed

Reviewer: Incidentally, I like this idea of adding the changelog entry on the PR with the change. Maybe we should start doing that on the libraries?

> - get_arg_or_config now correctly checks the config service if the passed
>   namespace contains the given value but it is None.

Reviewer 2: ```suggestion
- ``get_arg_or_config`` now correctly checks the config service if the passed
  namespace contains the given value and the value is ``None``.
```

Author: The latter part of that makes it more confusing IMO. What about "if the passed namespace has ``None`` as the value of the requested item."

Reviewer 2: Yup, that works too.
